HP-779: rebase develop branch to master

NikoHelle opened the pull request:
A lot have changed in develop-branch. Mostly in /src - meaning React files. Possible conflicts will probably be caused by changed config files. 

### Major changes in configs

**.github/workflows/*.yml**
- new props from development that did not exist in master
- using service port from master

**.prod/nginx.conf**
- file in almost 100% new. Changes most likely done for OpenShift.
- this has not been tested in KuVa environments

**scripts/env.sh**
**scripts/update-runtime-env.ts**
- new files in master
- these have not been tested in KuVa environments

**dockerfile**
- changed a lot, but can be built successfully

dependabot[bot] committed:
Bump http-proxy from 1.18.0 to 1.18.1

Bumps [http-proxy](https://github.com/http-party/node-http-proxy) from 1.18.0 to 1.18.1.
- [Release notes](https://github.com/http-party/node-http-proxy/releases)
- [Changelog](https://github.com/http-party/node-http-proxy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/http-party/node-http-proxy/compare/1.18.0...1.18.1)

Signed-off-by: dependabot[bot] <support@github.com>

Sukriva committed:
Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update nginx.conf

Update Dockerfile

Update nginx.conf

Update nginx.conf

Update nginx.conf

Update nginx.conf

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update Dockerfile

Update nginx.conf

Update nginx.conf

Update Dockerfile

Update nginx.conf

Jokinen committed:
Improve authentication error handling

Jokinen committed:
Refactor authentication

NikoHelle committed:
Updated required packages and their dependencies

Changed packages:

- yup had severe vulnerabilites

- react-scripts had most vulnerabilies and required major version update

- i18n needed update too and updated also other x-i18n packages for compatibility

- typescript was also required to update because of i18n and new "import type {...}

- lodash had a severe vulnerability

- added also eslint to devDependencies because multiple packages require it

- had to add eslint-config-react-app to dependencies because of yarn bug which installs it with react-scripts under react-script -folder where it is not found.

Other changes:

Removed eslintConfig from package.json because it is in .eslintrc

Changes in tsconfig.json are made by react-scripts

Fixed all error-level parsing errors in JS/TS files by adding rule overrides.

NikoHelle committed:
Fix lint errors introduced in react-script & typescript update

Global eslint changes in .eslintrc:

- no-use-before-define was added / fixed as stated in:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-use-before-define.md#how-to-use

- "no-undef": "error" was added to prevent compile errors.

Other changes in tsx/ts -files fix mostly linting errors when function return type is not defined or module exports something by default which is not named first.

Obsolete jest-localstorage-mock was removed because new version of Jest includes same features and they do not work together.

NikoHelle committed:
Fix broken tests

New localstorage mock from Jest caused spy-functions to break

Toast.tsx broke because it really didnt work before and updated Jest caused an error when trying to remove element that is not found.

NikoHelle committed:
Remove disabled linting and fix raised errors

Some linting errors where disabled with eslint-disable-line. Removed those and fixed errors caused by them.

Removed also one useless line that was commented out

iivorait committed:
Changed nginx version to UBI8/1.18

NikoHelle committed:
Fix typescript error while running docker-compose up. @typescript-eslint/ban-ts-comment not recognised

NikoHelle committed:
Fix ts compile error with useProfile.ts and missing prop error caused by outdated @types

joonvena committed:
Switch to Github Actions

NikoHelle committed:
Extend Toast interface in Toast React component

Component props almost matched Toast interface so extending the interface is better than defining same props again

Hi-Fi committed:
Dynamically creating client to KuVa Tunnistamo

Jira: DEVOPS-30

NikoHelle committed:
Move eslint*-* to dependencies to avoid docker build error

NikoHelle committed:
Removed useless eslint-config-react-app

It was added in one commit to prevent lint error that is no longer happening

Hi-Fi committed:
Configuration at runtime from environment variables

`envsubst` to correct variables that refer to other variables

Hi-Fi committed:
Scripts to use normal CRA env variables

Hi-Fi committed:
Local running using starttime environment configuration as deployed version

JIRA: DEVOPS-202

NikoHelle committed:
Change styles so it matches new layout

Removed some padding from lists

Hi-Fi committed:
Documentation updated

NikoHelle committed:
Updated translations

En / sv -translations are missing so placeholders added

NikoHelle committed:
Added linting rule 'curly' and fixed raised errors
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS121&rule_key=typescript%3AS121

NikoHelle committed:
Added linting rule 'consistent-return' and fixed raised errors
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript:S3801&rule_key=typescript:S3801

NikoHelle committed:
Added linting rule 'no-shadow' and fixed raised errors
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS1117&rule_key=typescript%3AS1117.

NikoHelle committed:
fix Sonar error The 'any' type should not be used

NikoHelle committed:
Installed eslint-plugin-sonarjs and added linting rule 'no-duplicate-string'
Fixed errors raised by new rule. Rule documentation https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS1192&rule_key=typescript%3AS1192

NikoHelle committed:
Added linting rule 'arrow-body-style' and fixed raised errors
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS1192&rule_key=typescript%3AS1192.
Also added rule 'arrow-parens' forcing arrow function format to be consistent and fixed raised errors.
Also fixed type errors with HDSOptionType which were shown only on compile

NikoHelle committed:
Added linting rule 'object-shorthand' and fixed raised errors
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS3498&rule_key=typescript%3AS3498.

NikoHelle committed:
Fix rule errors where both '?' and 'undefined' were used
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS4782&rule_key=typescript%3AS4782

NikoHelle committed:
Added linting rule '@typescript-eslint/no-unnecessary-type-assertion' and fixed raised errors
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS4325&rule_key=typescript%3AS4325. parserOptions in .eslintrc was a requirement for the rule to work.

NikoHelle committed:
Fixed errors for rule 'Default export names and file names should match'
Sonar rule: https://sonarcloud.io/organizations/city-of-helsinki/rules?open=typescript%3AS3317&rule_key=typescript%3AS3317. Also fixed type error discovered when starting app

NikoHelle committed:
Added missing dependency used just for typing

NikoHelle committed:
Fix missing lang, title and integration hash.

Two bugs and a security hotspot reported by Sonar

NikoHelle committed:
Fixed Sonar code smell 'Optional boolean parameters should have default value'

NikoHelle committed:
Moved sonarjs eslint package to dependencies to fix build error in CI

NikoHelle committed:
Fix codesmell 'Redundant casts and non-null assertions should be avoided'

Hi-Fi committed:
GHA CI

Jira: DEVOPS-210

Hi-Fi committed:
Fix service port to work with UBI image

Hi-Fi committed:
Env variables to work with Windows

NikoHelle committed:
Changed tests to use same, one time configured data

Tests had a lot of data defined multiple times. Matching data is now cloned when needed and set only once. Data objects are cloned because passing original objects can mess tests or allow them pass falsely.

NikoHelle committed:
Rename test data to improve clarity and create test data in cleaner way
Test data is simplified by cloning existing Profile data instead of declaring data multiple times. Naming data instead of using arrays and indexes makes tests easier to understand and follow.

Hi-Fi committed:
Resource requests and limits set

Jira: DEVOPS-231

NikoHelle committed:
Update HDS libraries

Switched to HDS navigation bar
Fullscreen navigation in included in HDS Navigation, so removed it.
ViewProfile menu moved to nav bar
UserDropdown and LanguageSwitcher use HDS components

NikoHelle committed:
Created a copy of HDS Notification component to fix a bug

HDS notification use react-spring which has a bug when building for production
https://github.com/pmndrs/react-spring/issues/1078

Created a copy and took off all animation features so react-spring is not needed.

NikoHelle committed:
Upadated apollo client and removed apollo-boost

Old client is depricated and does not support cache modifications. Apollo boost is not needed anymore

NikoHelle committed:
Include Verified personal information in GraphQLQuery.

Ignore error received when using non-verified IDP.

GraphQL data structure has been changed, so it was needed to run codegen that updates generatedTypes. Codegen does not work with current versions, because node_modules have multiple versions of GraphQL package. "Resolutions" in package.json resolves that with some package updates.

NikoHelle committed:
Render verified PI if received from BE.

Updated translation files.

NikoHelle committed:
Show suomi.fi as an AMR.

Also DownloadData and DeleteProfile components inside ProfileInformation cause unnecessary graphQL reloads when Profile changes. Moved their code to independent components to avoid reloading

NikoHelle committed:
Change primary heading to be h1 and fix nav zIndex problem with HDS

NikoHelle committed:
Test snapshot update

NikoHelle committed:
Added missing dependency

NikoHelle committed:
Small css and translation update

Missing translation added
Small css fixes..

NikoHelle committed:
Added new configuration variable documentation into README

NikoHelle committed:
Updated development configs to use OpenShift servers

NikoHelle committed:
Remove files related to subscriptions

Prettier made a change to "package.json"

NikoHelle committed:
Test update

Added tests for VerifiedProfileInformation. Requires mocked personal data, done in myProfileMocking.ts. Added new testing library for easier element queries from DOM.

Added suomi.fi test to ProfileInformationAccountManagementLinks.

NikoHelle committed:
One typings file for all graphQL type shortcuts

There started to be too many variations of same shortcuts (Address, AddressNode, ProfileAddress, etc.) so made one file to define them all.

Named all root nodes as ***Root, because for example MyProfile is too generic and there is MyProfile_profile and same thing with UpdateProfile, Subscriptions, MySubscriptions etc.

NikoHelle committed:
Centralized schemas and formProperties

Schemas were defined in multiple places, like properties of form inputs. Easier to validate, create error messages and use same props in all forms.

Previously to show min and max lengths in input errors required adding them to form input function calls. Now the are embeeded to formik error types. Unfortunately they cannot be extracted from yup objects.

setErrorMessageWithOptions adds props to formik error type: validation.maxLength becomes validation.maxLength?max=266&min=1
getErrorMessageWithOptions gets min and max from formik error type

Query format (?a=1&b=2) is used because js handles it with natively

EditProfile*.* will be removed so schemas/formProps not implemented there

NikoHelle committed:
Use formProps in forms

NikoHelle committed:
Improved Formik error rendering

Added oldData indicator so validation errors are not shown
while typing first letter of new data.

Created type for ErrorRenderer.

CreateProfile uses new error rendering without retyping min and max.

NikoHelle committed:
Fix bug in UpdateMutationVariables

Email was used as id.

NikoHelle committed:
(HP-318) useProfileQuery added

useProfileQuery handles profile loading. It uses lazyQuery, because profile should not be auto-loaded, because user might not be logged in when hook is used.

Added ahook library so passed onError-function is dependency safe and does not cause onError calls just if onError changed.

NikoHelle committed:
(HP-318) ProfileContext added

ProfileContext is a React context that is the only place MyProfile query result is stored. All components use it and only it to access user data. It uses useProfileQuery to fetch profile.

NikoHelle committed:
(HP-318) MockApolloClientProvider added

MockApolloClientProvider mocks responses from graphQL server. It receives a function that returns next response for a query. Usually only one response is needed, but UI testing requires multiple. Responses may be myProfile queries, update queries when mutating or errors. There are two error types: network and graphql.

resetApolloMocks is required to reset everything between tests. Cache prevails between tests in same test file, so it must be reset with resetStore.

jest-fetch-mock moved to dependencies or automatic linter in GitHub wont find it.

NikoHelle committed:
(HP-318) exposeProfileContext added to componentMocking

React hooks are exposed to tests with @testing-library/react-hooks. It can return all props returned from the hook.

useProfileContext required Apollo client so hook is wrapped with MockApolloClientProvider. And ProfileProvider to provide to hook to children.

waitForUpdate and waitForDataChange are helper functions for tests to wait for updates from graphQL queries. testing-library has waitForNextUpdate, but it does not update everytime.

NikoHelle committed:
(HP-318) Return copy of MyProfile from myProfileMocking

Removed old src/common/test/myProfileQueryData.ts that didnt return a copy of data

Fixed dependencies

NikoHelle committed:
(HP-318) Tests for useProfileContext hook

Test queries with the hook. Hook uses lazyLoading so at first there is no data and nothing is loading. fetch() triggers the first load and success and error responses are handled.

NikoHelle committed:
(HP-318) EditData.ts added

EditData picks editable props from MyProfile and creates an array of values that components use. EditData items are readonly, so they cannot be mutated. New items will be created on each mutation.

Properties:
- id: same id as in MyProfile
- value: depends on dataType. Components use this as default value(s) in forms. Value is picked from MyProfile
- primary: is the item primary.
- saving: hooks, data updates and components need to know if items are being saved and what for. Update listeners need to know what props to observe and for example UI must know if all items should be locked when saving new primary item

Added also tests for creating data.

NikoHelle committed:
(HP-318) ProfileContext has getName and errorListeners

As ProfileContext is the source for MyProfile it should also provide utility functions like getName()

Added also errorListeners. addErrorListener returns a disposer function that removes that listener.

useProfileErrorListener hook uses profileContext.addErrorListener but it also handles disposing when component is removed via useEffect hook. UseMemo is used so addListener is called only once. Using usePersistFn makes it safe to use changing functions as listeners and useMemo dependency. It stores latest version of the listener without invoking dependency change.

Added error tracking to componentMocking and unique errors to MockApolloClientProvider

NikoHelle committed:
(HP-318) Use new ProfileContext in Components

Added ProfileContext Provider to App.tsx and then the context can be consumed by child components.

ProfileInformation has only placeholder for new components as old ones do not use Context.

ViewProfile in never rendered if Profile (parent) does not have data so ViewProfile do not need fetch or error handling.

Added new color that is not in HDS.

NikoHelle committed:
(HP-318) Add test ids and mocks

Created mock files for DownloadData and DeleteProfile so queries in actual components wont be run, when testing parent components

Test ids added for easy element finding

Mocked Route did not have useLocation mock needed in Profile.tsx testing

NikoHelle committed:
(HP-318) ComponentMocking test improvements

Added BrowserRouter and ToastProvider because needed in Profile and ViewProfile tests

TestTools have helper functions waitforEl, waitForIsComplete and waitForDataChange so it is convenient to wait for data loading and re-rendering in tests

ProfileContextAsHTML component renders the profile context  as HTML and makes it easy to read updates. Hooks are not exposed like in exposeProfileContext and better to use HTML as source with Components than make work-arounds for hooks.

NikoHelle committed:
(HP-318) Profile.test.tsx created

Tests for Profile. Had to remove useHistory mock from setUpTests or route changes could not be tested.

NikoHelle committed:
(HP-318) ViewProfile.test.tsx created

NikoHelle committed:
(HP-318) useProfileDataEditor created

Hook for using editData and (later) modify it

NikoHelle committed:
(HP-318) BasicData component created

Component for rendering and (later) editing basic data.

Added it into ProfileInformation

NikoHelle committed:
(HP-318) BasicData tests

Check user data is rendered.

Component cannot be rendered without data, so added a new helper
component to componentMocking:
RenderChildrenWhenDataIsComplete

NikoHelle committed:
(HP-318) Remove old editProfile files

These files are not needed anymore.

Fix dependencies in files with dependency to removed files.

NikoHelle committed:
Added expiration check to getUser

When user has left session open without logout and returns, the userManager gets user from the localStorage without checking if the user has expired. After that fetching user's profile will return an error. Fetching profile is useless and now avoided by checking also user.expired instead of just existence of user.

NikoHelle committed:
(HP-318) Update getEmails/Phones/Addresses to keep primary

NikoHelle committed:
(HP-318) Update test data to have correct MyQuery structure

NikoHelle committed:
(HP-318) updateMutationVariables to support partial updates

NikoHelle committed:
(HP-318) updateMutationVariables to support partial updates

NikoHelle committed:
(HP-318) Update jest-fetch-mock
Now MockApolloClient can return query variables

NikoHelle committed:
(HP-318) Add AnyObject type to typings

NikoHelle committed:
(HP-318) Add all profile data to UpdateMyProfile query
This way cache can be fully updated without a new MyProfile query
Generated also generatedTypes.ts

NikoHelle committed:
(HP-318) Add all profile data to UpdateMyProfile query
This way cache can be fully updated without a new MyProfile query
Generated also generatedTypes.ts

NikoHelle committed:
(HP-318) Add hook for mutations

The main purpose of this commit is "useProfileMutations.ts". It sends profile data updates to the server and updates cache after that. It uses ProfileContext and also passes through its data, so "useProfileDataEditor" is required to use only useProfileMutations and not also useProfileContext.

Testing mutations require a lot of new testing tools:
- "myProfileMocking.ts" has "cloneAndManipulateProfile" that provides functions for easy profile editing. It outputs new profile and also formValues needed to mock updates. New profile data is then used in mock responses and formValues when creating update variables.
- "componentMocking.ts" exposes the "useProfileMutations" hook for testing. Added also simple "createResultPropertyTracker" that creates a promise based listener for hook's properties.  "ProfileContextFetcher" component just auto-fetches profile data for the hook.

Also bundled all clean-up functions for tests to "cleanComponentMocks" function and updated ProfileContext to use it like the new UseProfileMutations.tests.ts

useProfileDataEditor hook now uses useProfileMutations instead of useProfileContext

NikoHelle committed:
(HP-318) Add await-to-js
Makes async/await error handling easier

NikoHelle committed:
(HP-318) Added data updating to editData.ts

Renamed 'create' to 'createNewItem' to make it cleared what it creates

New functions:
- updateItemAndCreateSaveData: called when saving. It updates passed item by cloning it and all other items too. Value of the cloned item is set as passed new value. Function returns a formValue object that is passed on to updateMutationVariables which creates the update variables. While saving the item may not change and "preventDoubleEdits" -function checks that.

- updateData: called after save. If editData is waiting for an update (any item.saving is set). Items are updated if new profile data has same value edit data does.

-updateAfterSavingError: when save error occurs, item.saving is set to "undefined"

-updateItemAndCloneList: clones item, updates clone.value and returns all items as new, cloned list

-hasItemUpdated: returns true/false, depending on whether prop that is being saved (value/primary) matches the same prop in profile

-updateItems: updates items being saved if hasItemUpdated() returns true. Tracks also the new item (even they cannot be created yet). There can be only one new item at a time, so once it's found in profile data, no need to check new items anymore. Returns new list only if any items where updated.

NikoHelle committed:
(HP-318) Added saving to useProfileDataEditor

useProfileDataEditor now uses new update functions in editData and useProfileMutations to update profile

NikoHelle committed:
(HP-318) Editing and saving added to BasicData.tsx

Basic data can now be edited.

New components:

- SaveIndicator.tsx. Notifies screen readers that save is in progress
- EditingNotifications.tsx Notifies all users when saving in complete (success/error)
- useNotificationContent.ts creates content for EditingNotifications.tsx and makes it more general and easier to update.
- EditButton.tsx renders buttons for making changes (edit/remove/set-primary)
- FormButtons.tsx renders save/cancel buttons when editing

Updates
- componentMocking has a lot of new helpers to test components and UI-flow.
- BasicData.test.tsx updated of course

NikoHelle committed:
(HP-318) Cancel-button resets to previous version

When user edits an item, tries to save it and it fails, item.value should be reset to previous value. Added storage for previous versions: backups. Before storing new value, old one is backed up and restored with reset().

NikoHelle committed:
(HP-318) Fix wrong button position.

NikoHelle committed:
(HP-318) Unify test files - always use "it()"
Some tests used test(), some it(). Now all use it()

NikoHelle committed:
(HP-318) Ran prettier for all .css files

For some prettier extension issue, css files had never been run through prettier.

NikoHelle committed:
(HP-318) Identical functions merged
myProfileMocking had identical function with EditData.
Removed from myProfileMocking.

NikoHelle committed:
(HP-318) Removed useless testId from BasicData

NikoHelle committed:
(HP-318) Decrease test data repetition

Common code grouped as one and removed useless const

SQ basicdata.test. update

NikoHelle committed:
(HP-318) Renamed cloneAndManipulateProfile to
cloneProfileAndProvideManipulationFunctions

NikoHelle committed:
(HP-318) Renamed "triggerAction" to "clickElement" in tests

NikoHelle committed:
(HP-318) Renamed test function to describe it better

renderProfileContextWrapper renamed
 to renderComponentWithMocksAndContexts

NikoHelle committed:
(HP-318) Add and remove items added to EditData.ts

New items can be added and removed. There is no button yet in the UI for removing old data. New item is removed when cancel is pressed.

Added id to createFormValues(), otherwise all items are always removed and re-created when saving.

NikoHelle committed:
(HP-318) Add and remove item to useProfileDataEditor

Only new item removal is supported in this commit

NikoHelle committed:
(HP-318) Common error messaging added to BasicData.tsx

All components need same hasFieldError and getFieldErrorMessage -functions, so added them to formik.ts

NikoHelle committed:
(HP-318) Check invalid values in BasicData.test.tsx

This also required update to componentMocking.tsx

ClickElement returns HTMLElement that was pressed. Element's "disabled"-property needs sometimes checking.

submit() has new parameter: optionalSubmitButtonSelector, because email, phones and addresses components will introduce different submit button selectors.

new functions for element attribute checking.

waitForElementAndValue() moved above its usage in setInputValue()

NikoHelle committed:
(HP-318) Render and edit addresses, emails and phones

Components for handling multiItemProfileData.

MultiItemEditor handles rendering of item lists  and also handles common actions like saving and renders save indicators. It also shows button for adding new items.

MultiItemRow renders emails and phones, because those have same number of fields and are simple to handle.

MultiItemAddressRow is almost identical but different enough to have its own component for handling addresses

MultiItem*Row are like BasicData.tsx, but also handle new items.

useCommonEditHandling is used by MultiItem*Row because they have identical edit functions.

Remove and set-primary -buttons are not rendered at the moment.

NikoHelle committed:
(HP-318) Update editData updateItems test for all dataTypes

NikoHelle committed:
(HP-318) Fix bug in useProfileMutations.test.ts

NikoHelle committed:
Change local to sessionStorage in authService

NikoHelle committed:
Remove useless localStorage listener in App.tsx

NikoHelle committed:
Change local to sessionStorage in GDPR downloads

NikoHelle committed:
Update oidc-client version to 1.11.5

NikoHelle committed:
Change correct profile api url to .env

NikoHelle committed:
Changed yle-amr to yletunnus-amr

Translations updated too because amr is a translation key

NikoHelle committed:
(HP-318) Added old item removal to useProfileDataEditor.ts

NikoHelle committed:
(HP-318) Update ConfirmationModal
New Styles and added test ids

NikoHelle committed:
(HP-318) Add modal container to test DOM

NikoHelle committed:
(HP-318) Removal to UI

Biggest change here is the useConfirmationModal which makes modals async/await compatible. Some extra code, but anyway all modal props and state handling (open / close) would be added to EditButtons.tsx.

Enabled remove buttons in UI.

NikoHelle committed:
(HP-318) EditData.test.ts minor boolean test change

NikoHelle committed:
(HP-318) Changed postalCode max length to 32

NikoHelle committed:
(HP-318) Split componentMocking.tsx into smaller files

componentMocking.tsx grew too much and tried to do much more than name suggests. Instead of importing componentMocking.tsx, now use testingLibraryTools.ts

React components are moved their own files
ExposeHooksForTesting.tsx has hook test utils
renderComponentTestDOM creates the DOM with mockProviders and contexts.

No new code added, just splitted and removed componentMocking.tsx.

NikoHelle committed:
(HP-318) Added setPrimary to edit functions

Added unit tests for setPrimary

NikoHelle committed:
(HP-318) Enable setPrimary in UI

All edit buttons must be disabled while reordering items
to prevent data loss.

NikoHelle committed:
(HP-318) Added profile language selection

NikoHelle committed:
(HP-318) Accessibility: return focus to previous element

When dynamically changing UI, like showing a form and hiding it after save, the focus should return to the element that initiated the change, so keyboard users can continue from there.

Focus is now returned to edit or 'add new' button, when action is done or cancelled.

NikoHelle committed:
(HP-318) Accessible action labels

Buttons in EditButton component now have informative aria-labels that include the value the button targets. For example
"Remove/Edit email <the email>".

NikoHelle committed:
(HP-318) Accessibility improvment with focus keeper

When a form is rendered, focus should be kept inside the form until it is saved or cancelled. Keeping focus inside prevents screen reader users to get lost on the page and wander off the form.

NikoHelle committed:
(HP-318) Remove unused files

NikoHelle committed:
(HP-318) Get latest item version in EditData

UI might pass old version of data item. Always get the latest version with item.id

NikoHelle committed:
(HP-318) Fixed code smell

NikoHelle committed:
(HP-318) Remove unused subscriptions path

Merging HP-318 brought back unused path removed in HP-548

NikoHelle committed:
(HP-318) Render Verified Personal information

Verified personal information component was not used. Updated the component to use ProfileContext so it matches BasicData and their usage is same in ProfileInformation.tsx.
Test file needed an update too because of the ProfileContext change.

NikoHelle committed:
(HP-819) Added municipality and updated translations

NikoHelle committed:
(HP-819) Fixed bg color of service connections page

NikoHelle committed:
(HP-819) Removed useless user link and added new test ids

NikoHelle committed:
(HP-727) Update packages

NikoHelle committed:
(HP-727) Change props "label" and "helperText"

labelText does not exist anymore.

helperText is only used with errors, so errorText is the correct prop.

NikoHelle committed:
(HP-727) Replace deprecated Dropdown

Select-component is used instead of Dropdown. If option list is long (countries in addresses) Combobox is used, because it has search functions. Side-effect of search is possibility to unselect. So added error handling to countries.

It is a known bug that Combobox error message  has different icon than textboxes.

Added label back to 'Asiontikieli' because it improves accessibility. Element is hidden with css, but aria-label uses the label prop as a prefix.

getA11yStatusMessage reads the selected item as "Suomi valittu" or "Maa ei ole valittuna".

NikoHelle committed:
(HP-727) Updated test snapshots

NikoHelle committed:
(HP-727) Use HDS body font-size and line-height

NikoHelle committed:
(HP-727) Use hds font-sizes instead of px

NikoHelle committed:
(HP-727) Match label and value font-sizes

Text element label/value font-sizes should match with input label/value.

This way elements won't "jump" when changing edit mode.

NikoHelle committed:
(HP-727) Removed common/Button and use hds/Button

NikoHelle committed:
(HP-727) Hide icon from screen reader
The svg image has no title and is read as "unlabeled image"

NikoHelle committed:
(HP-727) Use HDS Accordion hook in Expanding Panel

Accordion hook used instead of Accordion component, because it cannot be initially open.

Removed keyboard controls because they for example prevent toggling checkboxes with keyboard.

NikoHelle committed:
(HP-727) Uppercase translations converted to lower

Uppercase translations are needlessly read in different way when using Voice over.

NikoHelle committed:
(HP-727) Remove unused css module

NikoHelle committed:
(HP-727) Remove unused styles

NikoHelle committed:
(HP-727) Use HDS Dialog

ConfirmationModal and BerthErrorModal use the Dialog instead of ReactModal. Renamed some props: modalTitle to title and modalText to content.

Made ConfirmationModal simpler by separating service list rendering to its own component DeleteProfileContent.

Removed css-files related to modals which became obsolete.

NikoHelle committed:
(HP-727) Removed react-modal

NikoHelle committed:
(HP-727) Use HDS PhoneInput with phone numbers

NikoHelle committed:
(HP-727) Removed fixes that are fixed in HDS

NikoHelle committed:
(HP-727) Removed AccessibilityShortcuts

HDS navigation has same function built-in

NikoHelle committed:
(HP-727) Remove background color from verifiedData section

NikoHelle committed:
(HP-727) Rename tsx-file's own css classNames to dash-cased

Dev.hel.fi coding standards require this.

Removed also styles that are not used anymore

NikoHelle committed:
(HP-727) Rename classNames in form.module.css to dash-cased

Dev.hel.fi coding standards require this.

NikoHelle committed:
(HP-727) Rename classNames in responsive.module.css

ClassNames changed to  to dash-cased
Dev.hel.fi coding standards require this.

Removed unused classes.

NikoHelle committed:
(HP-727) Fix conflicting text messages

UI showed "you have following services" and "No Services" -texts simultaneously when there were no services.

NikoHelle committed:
(HP-727) Update version

NikoHelle committed:
(HP-727) Fix layout bug with visually hidden element
If element has width:100% and class visually-hidden, its size can be > 100% and horizontal scroll bar appears.

NikoHelle committed:
(HP-727) Added translated appName to Header

NikoHelle committed:
(HP-652) Improve language selections

Copied aria-labels from HDS examples.
Added lang-attribute to options
Fixed bug with mobile view by adding componentName
Logo image is changed with the language

NikoHelle committed:
(HP-652) Fixed login button focus color
The button was invisible.

NikoHelle committed:
(HP-652) Better accessibility when link opens in new window

Links that open to a new window, should inform it in aria-label. Added new component that creates the label and adds other attributes.

NikoHelle committed:
(HP-652) Labels should not have html
Moved HTML outside of labels as a paragraph and changed label text

NikoHelle committed:
(HP-652) Remove multiple h1 tags
Instead of h1s with class names to make font smaller, use real h2/h3 tags.

NikoHelle committed:
(HP-652) Removed titleButton prop from ProfileSection

It is not used anywhere

NikoHelle committed:
(HP-652) List all form errors in an alert
Create an alert with combined error message for screenreaders when a form is submitted.
Changed 'country' to 'countryCode' in formProperties, so it matches the component's data props.

NikoHelle committed:
(HP-652) Added aria helpers for inputs
More informative aria-labels with aria-labelledby -linking from the input to a dom element. Using elements instead of attributes, because otherwise same labels would be created and repeated in all input fields in phones, addresses and emails.

NikoHelle committed:
(HP-652) Add home link to Logo and title

NikoHelle committed:
(HP-652) Change page title when new route is rendered
Added a PageMeta-component that changes document.title

Removed AppMeta because PageMeta does the same
Fixed non-existing translation key

NikoHelle committed:
(HP-652) Fix tests with errorSelectors
The test "expect(() => errorElementGetter).not.toThrow();" will never throw.

Changed to expect(errorElementGetter).not.toThrow();

That's why tests did not fail even if error selector did not find element.

NikoHelle committed:
(HP-652) Return focus to remove-button when cancelled

NikoHelle committed:
(HP-652) Set focus back to delete button when cancelled

NikoHelle committed:
(HP-652) Remove berthError handling from ProfileInformation

Error is handled in DeleteProfile.tsx

NikoHelle committed:
(HP-652) Version pump

NikoHelle committed:
(HP-652) Fix empty headings
Added content check up

NikoHelle committed:
(HP-652) Fix "Must have one main element"

NikoHelle committed:
(HP-652) Make footer logo a link to home page

NikoHelle committed:
(HP-652) Hide icon from screen reader
Fixed missing alt-text warning

NikoHelle committed:
(HP-652) Added jsx-a11y eslint plugin
Fixed "non-interactive element should not have focus" -errors

Disabled "jsx-a11y/no-autofocus" because autofocus was not mentioned in accessibility audit and it is a convenient feature

NikoHelle committed:
(HP-652) Change logo link translations
Logos are links to homepage, so translations should inform that

NikoHelle committed:
(HP-652) Match CreateProfile styling to other forms
The form used its own styling which differs slightly from other styles.
The email field uses readonly TextInput so it is aligned properly

NikoHelle committed:
(HP-834) Update react-scripts to 4.0.3.

NikoHelle committed:
(HP-775) Upgraded validator and its types

NikoHelle committed:
(HP-774) Upgraded lodash package and types

NikoHelle committed:
(HP-773) Upgrade i18next and related packages

NikoHelle committed:
(HP-622) Updated graphQL

Ran codegen which updates generatedTypes.ts.
Then fixed typings.ts accordingly.

NikoHelle committed:
Re-installed dependencies and cleaned yarn lock file
Yarn did not seem to dedupe dependencies correctly and left multiple versions of packages that have mutual version requirements
Removed yarn.lock (and node_modules) and ran "yarn".

NikoHelle committed:
(HP-585) Add response_type as env parameter

NikoHelle committed:
(HP-585) Update README

NikoHelle committed:
(HP-585) Removed unused config props
While adding the env.response_type I found out deleted props are not used anywhere

NikoHelle committed:
(HP-621) Removed translations related to subsciptions

NikoHelle committed:
(HP-791) Add new helper for parsing url params

NikoHelle committed:
(HP-791) Add error page and route

NikoHelle committed:
(HP-791) Redirect login errors to error page
Added useErrorPageRedirect-hook to make it easier to redirect to error page.

NikoHelle committed:
(HP-791) Added ErrorPage tests

NikoHelle committed:
(HP-791) Stylize the "Authenticating"-message
Plain text looked like a styling bug. Added spinner and centering.

NikoHelle committed:
(HP-779) Version bump

NikoHelle committed:
(HP-779) Revert prod and review service port to 80

NikoHelle committed:
(HP-779) Match nginx server port to yml server port

Jokinen commented:
Review environment available here: https://open-city-profile-ui-hp-779-merge-develop-be.test.kuva.hel.ninja/login

Currently logging in does not work--as far as I can remember it has worked before.

I wonder if this value has fallen out of date: 
https://github.com/City-of-Helsinki/open-city-profile-ui/blob/develop/.github/workflows/review.yml#L54

Seems like it's possible that either `-ui` or `-ui-dev` postfix should be used 🤔
https://github.com/City-of-Helsinki/open-city-profile-ui/search?q=https%3A%2F%2Fapi.hel.fi%2Fauth%2Fhelsinkiprofile

NikoHelle committed:
(HP-779) Change exposed port in docker to 80

NikoHelle committed:
(HP-779) Moved env vars from build to deploy

NikoHelle committed:
(HP-779) Add production env vars

Overwrote .env response_type and kept it as it was in prod: 'id_token token'

Some vars are same as in .env, but it is clearer to add them in yml.

NikoHelle committed:
(HP-779) Add staging env vars

Some vars are same as in .env, but it is clearer to add them in yml.

Jokinen reviewed (approved):
Review environment works and the configs for tother envs look good 👍

NikoHelle committed:
(HP-779) Update translations

NikoHelle committed:
(HP-779) Change allowed Tunnistamo response_types to "code"

Set UI response_type to code.

(HP-779) SQ prev

NikoHelle commented:
Closing because newer version was merged to master.
